osu.Game/Screens/Edit/Compose/Components/SelectionBoxRotationHandle.cs

```diff
@@ -127,7 +127,9 @@ private float convertDragEventToAngleOfRotation(DragEvent e)
         private void applyRotation(bool shouldSnap)
         {
             float newRotation = shouldSnap ? snap(rawCumulativeRotation, snap_step) : MathF.Round(rawCumulativeRotation);
-            newRotation = (newRotation - 180) % 360 + 180;
+            newRotation = ((newRotation + 360 + 180) % 360) - 180;
```

Comment:
`newRotation` is post-processed `rawCumulativeRotation` which is a difference of two `atan2()` calls. The result of `atan2` is always in the interval $[-\pi, \pi]$, so adding $2\pi$ does ensure a positive number which is safe to pass to the remainder operator is attained. The 180 on top is to get a range of [-180, 180] rather than [0, 360] from the inner operation and is correctly undone by the subtraction after the remainder.